GoogleForCreators/web-stories-wp
# Disable cross-origin isolation for contributors

## Context



Cross-origin isolation is currently needlessly enabled even if user cannot upload media.

## Summary



This PR adds a capability check to short-circuit this logic.

## Relevant Technical Choices



Updated `Cross_Origin_Isolation::is_needed()`, extended tests.

## To-do



## User-facing changes



N/A

## Testing Instructions



### QA






This PR can be tested by following these steps:

1. Log in as contributor user
2. Open editor
3. Check `window.crossOriginIsolated` value in console, see that it's `false`

### UAT







This PR can be tested by following these steps:

1.

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #8163
